homeassistant/components/ecovacs/__init__.py

```diff
@@ -1,91 +1,120 @@
-"""Support for Ecovacs Deebot vacuums."""
-import logging
-import random
-import string
+"""Support for Ecovacs vacuums."""
+import asyncio
+from functools import partial
 
 from sucks import EcoVacsAPI, VacBot
-import voluptuous as vol
 
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, EVENT_HOMEASSISTANT_STOP
-from homeassistant.helpers import discovery
-import homeassistant.helpers.config_validation as cv
-
-_LOGGER = logging.getLogger(__name__)
+from homeassistant.const import (
+    CONF_DEVICE_ID,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    EVENT_HOMEASSISTANT_STOP,
+)
 
-DOMAIN = "ecovacs"
+from .const import (
+    CONF_CONTINENT,
+    CONF_COUNTRY,
+    DATA_REMOVE_LISTENER,
+    DEVICES,
+    DOMAIN,
+    ECOVACS_ATTR_DEVICE_ID,
+    ECOVACS_ATTR_NAME,
+    LOGGER,
+    PLATFORMS,
+)
 
-CONF_COUNTRY = "country"
-CONF_CONTINENT = "continent"
 
-CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_USERNAME): cv.string,
-                vol.Required(CONF_PASSWORD): cv.string,
-                vol.Required(CONF_COUNTRY): vol.All(vol.Lower, cv.string),
-                vol.Required(CONF_CONTINENT): vol.All(vol.Lower, cv.string),
-            }
-        )
-    },
-    extra=vol.ALLOW_EXTRA,
-)
+async def async_setup(hass, config):
+    """Set up the Ecovacs integration."""
+    return True
```

Comment:
To avoid the breaking change I think you should allow the user to automatically migrate to the config entry and put and invalidate configuration + 3 releases